npm/networkPolicyController_test.go

```diff
@@ -0,0 +1,281 @@
+// Copyright 2018 Microsoft. All rights reserved.
+// MIT License
+package npm
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/Azure/azure-container-networking/npm/ipsm"
+
+	corev1 "k8s.io/api/core/v1"
+	networkingv1 "k8s.io/api/networking/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	kubeinformers "k8s.io/client-go/informers"
+	k8sfake "k8s.io/client-go/kubernetes/fake"
+	core "k8s.io/client-go/testing"
+)
+
+type netPolFixture struct {
+	t *testing.T
+
+	kubeclient *k8sfake.Clientset
+	// Objects to put in the store.
+	netPolLister []*networkingv1.NetworkPolicy
+	// (TODO) Actions expected to happen on the client. Will use this to check action.
+	kubeactions []core.Action
+	// Objects from here preloaded into NewSimpleFake.
+	kubeobjects []runtime.Object
+
+	// (TODO) will remove npMgr if possible
+	npMgr            *NetworkPolicyManager
+	ipsMgr           *ipsm.IpsetManager
+	netPolController *networkPolicyController
+	kubeInformer     kubeinformers.SharedInformerFactory
+}
+
+func newNetPolFixture(t *testing.T) *netPolFixture {
+	f := &netPolFixture{
+		t:            t,
+		netPolLister: []*networkingv1.NetworkPolicy{},
+		kubeobjects:  []runtime.Object{},
+		npMgr:        newNPMgr(t),
+		ipsMgr:       ipsm.NewIpsetManager(),
+	}
+
+	f.npMgr.RawNpMap = make(map[string]*networkingv1.NetworkPolicy)
+	return f
+}
+
+func (f *netPolFixture) newNetPolController(stopCh chan struct{}) {
+	f.kubeclient = k8sfake.NewSimpleClientset(f.kubeobjects...)
+	f.kubeInformer = kubeinformers.NewSharedInformerFactory(f.kubeclient, noResyncPeriodFunc())
+
+	f.netPolController = NewNetworkPolicyController(f.kubeInformer.Networking().V1().NetworkPolicies(), f.kubeclient, f.npMgr)
+	f.netPolController.netPolListerSynced = alwaysReady
+
+	for _, netPol := range f.netPolLister {
+		f.kubeInformer.Networking().V1().NetworkPolicies().Informer().GetIndexer().Add(netPol)
+	}
+
+	f.kubeInformer.Start(stopCh)
+}
+
+func (f *netPolFixture) ipSetSave(ipsetConfigFile string) {
```

Comment:
with this, we will need IpTableSave and IpTableRestore, since each test will be manipulating iptable rules, we should clean them up once done

Reply:
Good point. Added two methods for iptables management. Now the methods are not used in networkPolicyController_test.go due to npm_test.go. We will use them later.